src/runtime/powerpc64.rs

```diff
@@ -0,0 +1,92 @@
+//! Run-time feature detection on PowerPC64.
+use super::{bit, linux};
+
+#[macro_export]
+#[doc(hidden)]
+macro_rules! __unstable_detect_feature {
+    ("altivec") => {
+        $crate::vendor::__unstable_detect_feature($crate::vendor::__Feature::altivec{})
+    };
+    ("vsx") => {
+        $crate::vendor::__unstable_detect_feature($crate::vendor::__Feature::vsx{})
+    };
+    ("power8") => {
+        $crate::vendor::__unstable_detect_feature($crate::vendor::__Feature::power8{})
+    };
+    ($t:tt) => { compile_error!(concat!("unknown PowerPC target feature: ", $t)) };
+}
+
+/// PowerPC CPU Feature enum. Each variant denotes a position in a bitset
+/// for a particular feature.
+///
+/// PLEASE: do not use this, it is an implementation detail subject to change.
+#[doc(hidden)]
+#[allow(non_camel_case_types)]
+#[repr(u8)]
+pub enum __Feature {
+    /// Altivec
+    altivec,
+    /// VSX
+    vsx,
+    /// Power8
+    power8,
+}
+
+pub fn detect_features<T: linux::FeatureQuery>(mut x: T) -> usize {
+    let mut value: usize = 0;
+    {
+        let mut enable_feature = |f| {
+            if x.has_feature(&f) {
+                value = bit::set(value, f as u32);
+            }
+        };
+        enable_feature(__Feature::altivec);
+        enable_feature(__Feature::vsx);
+        enable_feature(__Feature::power8);
+    }
+    value
+}
+
+/// Probe the ELF Auxiliary vector for hardware capabilities
+///
+/// The values are part of the platform-specific `asm/hwcap.h`
+impl linux::FeatureQuery for linux::AuxVec {
+    fn has_feature(&mut self, x: &__Feature) -> bool {
+        use self::__Feature::*;
+        match *x {
+            altivec => self.lookup(linux::AT::HWCAP)
+                .map(|caps| caps & 0x10000000 != 0)
+                .unwrap_or(false),
+            vsx  => self.lookup(linux::AT::HWCAP)
+                .map(|caps| caps & 0x00000080 != 0)
+                .unwrap_or(false),
+            power8 => self.lookup(linux::AT::HWCAP2)
+                .map(|caps| caps & 0x80000000 != 0)
+                .unwrap_or(false),
+        }
+    }
+}
+
+// TODO: This is a dummy linux on ppc does not provide this information
+// the string values match the output of LD_SHOW_AUXV=1 /bin/true on glibc
+// systems.
+impl linux::FeatureQuery for linux::CpuInfo {
+    fn has_feature(&mut self, x: &__Feature) -> bool {
+        use self::__Feature::*;
+        match *x {
+            altivec => self.field("Features").has("altivec"),
+            vsx => self.field("Features").has("vsx"),
+            power8 => self.field("Features").has("arch_2_07"),
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    #[test]
```

Comment:
Can you add a test that parses one or two dumps of `/proc/cpuinfo` for different CPUs?

Reply:
No, since the cpuinfo is w/out information at least for now.

Reply:
Check this out: https://github.com/randombit/cpuinfo/tree/master/ppc

That's a collection of `cpuinfo` files for a lot of powerpc cpus. Maybe one could c&p one of those files here and add it as a test?

Reply:
I'm afraid the comment is not clear in the function: `/proc/cpuinfo` on powerpc does not provide useful information, I left that code there since I plan to ask the linux kernel developers to provide the same information as arm does. I can change the function with a simple return false so it is less misleading for not.

Reply:
The comment is clear, I just skimmed too fast through it. 

What is the value of adding the cpuinfo implementation? We can always add it later once a kernel is released that "supports" it right?

Reply:
I just prefer to leave the placeholder since it is sort of future proof.

Reply:
Does the `linux` mod require this implementation to exist? If that is the case I think we should fix that instead. 

I can do that later if you prefer, but I think that leaving it with the expectation that a future Linux kernel version implements this is a long shot.

Reply:
The `linux` mod requires the implementation and I wouldn't spend more time on that. I updated it so it leverages the small bit of information to get to know if at least 1 feature is enabled.

Reply:
Now that the examples of cpuinfo for powerpc are there everything has become clear (sorry that it took so long). Just to check that I understand correctly, the only thing one can be checked from `/proc/cpuinfo` for PowerPC is whether `altivec` is supported or not, is that right?